### DISCUSSION TITLE
Multiple clients on the RS485 bus

### DISCUSSION THREAD

**Discussion by MrWaloo** [pymodbus-dev/pymodbus#1535] [Q&A] [answered]
Hi,

On one equipment, there is only one RS485 bus on which there already are one client and one server. The existing client polls every X seconds and then waits for Y seconds. I would like to know if there is a clean solution to wait until the bus is free in order to send requests during the idle time.
For now there is a workaround in a `while True` loop, for example:
```python
while True:
  time.sleep(1)
  rr = client.read_holding_registers(some parameters)
  if not rr.isError():
    break

decoder = BinaryPayloadDecoder.fromRegisters(rr.registers)
...
```
But to me this solution isn't *proper*.

Would there be a better solution?

Regards

**Comment by janiversen** [comment, marked as answer]
No there are no clean solution, because the modbus protocol is designed to have only one client (called master in the protocol spec).

**Reply by MrWaloo** [reply]
Thank you Jan
